src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -716,12 +718,20 @@ def prepare_inputs_for_generation(self, input_ids, past_key_values=None, attenti
         if attention_mask is None:
             attention_mask = input_ids.new_ones(input_shape)
 
-        return {
-            "input_ids": input_ids,
-            "attention_mask": attention_mask,
-            "position_ids": position_ids,
-            "past_key_values": past_key_values,
-        }
+        if inputs_embeds is not None and past_key_values is None:
```

Comment:
```suggestion
        # if `inputs_embeds` are passed, we only want to use them in the 1st generation step
        if inputs_embeds is not None and past_key_values is None:
```

(missing this comment, which I find helpful to understand what's going on in the if/else :) )